tests/test_register.py

```diff
@@ -318,3 +321,32 @@ def test_variable_expansion_permissions_config_style(self):
         assert config["permissions"][1]["service"] == "test-service-2"
         assert config["permissions"][1]["permission"] == "getcapabilities"
         assert config["permissions"][1]["group"] == admins
+
+    def test_get_all_config_from_dir(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            with tempfile.NamedTemporaryFile(mode="w", suffix=".cfg", dir=tmp_dir) as tmp1, \
+                 tempfile.NamedTemporaryFile(mode="w", suffix=".cfg", dir=tmp_dir) as tmp2:
+                # format doesn't matter
+                tmp1.write(json.dumps({"permissions": [{"perm": "permission1"}, {"perm": "permission2"}]}))
+                tmp1.seek(0)  # back to start since file still open (auto-delete if closed)
```

Comment:
maybe what you want is `tmp1.flush()`?

Reply:
maybe does the same, but using `seek(0)` as defined in tempfile doc